src/main/java/org/mongojack/PrePersist.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2011 VZ Netzwerke Ltd
+ * Copyright 2014 devbliss GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.mongojack;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation to support invoking a method on an entity before it is saved. Useful to support things like
+ * created/modified date setting, JSR-303 support, ...
+ *
+ * @author dnebinger
+ */
+@Documented
+@Inherited
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.METHOD})
+public @interface PrePersist {
```

Comment:
There's nothing to block using the one from JPA, I'm just not sure if there is any interference with JPA by reusing it.  For example, would it play well with Spring Data or JPA or ...?  Without going through every possible scenario, I couldn't be sure.  Morphia, for example, also declares and uses their own annotation rather than using the JPA one.

Reply:
At least in the early days of Morphia, the code was terrible. I haven't looked at it recently, but I wouldn't assume that they ran into problems using the JPA annotations. Given my past experience with Morphia, I think it's just as likely that they didn't put a lot of thought into how they wrote the code. I'm not suggesting we try to make MongoJack work with Spring Data or JPA at the moment, but using JPA annotations will make the path towards utilizing MongoJack with the rest of the Java ecosystem a little shorter.

Reply:
You don't have to make a new PR. You can just update this one